### ISSUE TITLE
Vulnerability Roundup 28

### ISSUE THREAD

**Issue by ckauhaus** [NixOS/nixpkgs#30959]
Auto-generated from `vulnix $( nix-instantiate -I nixpkgs=. '<nixpkgs/nixos/tests/login.nix>' )` on current master.

### audiofile-0.3.6 ([search](https://search.nix.gsc.io/?q=audiofile&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=audiofile+in%3Apath&type=Code))
 - [x] [CVE-2017-6827](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6827)
 - [x] [CVE-2017-6828](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6828)
 - [x] [CVE-2017-6829](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6829)
 - [x] [CVE-2017-6830](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6830)
 - [x] [CVE-2017-6831](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6831)
 - [x] [CVE-2017-6832](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6832)
 - [x] [CVE-2017-6833](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6833)
 - [x] [CVE-2017-6834](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6834)
 - [x] [CVE-2017-6835](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6835)
 - [x] [CVE-2017-6836](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6836)
 - [x] [CVE-2017-6837](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6837)
 - [x] [CVE-2017-6838](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6838)
 - [x] [CVE-2017-6839](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6839)

### cyrus-sasl-2.1.26 ([search](https://search.nix.gsc.io/?q=cyrus-sasl&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=cyrus-sasl+in%3Apath&type=Code))
 - [x] [CVE-2013-4122](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2013-4122)

### freetype-2.7.1 ([search](https://search.nix.gsc.io/?q=freetype&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=freetype+in%3Apath&type=Code))
 - [x] [CVE-2017-8105](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-8105)

### jquery-ui-1.11.4 ([search](https://search.nix.gsc.io/?q=jquery-ui&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=jquery-ui+in%3Apath&type=Code))
 - [x] [CVE-2016-7103](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-7103)

### libarchive-3.3.2 ([search](https://search.nix.gsc.io/?q=libarchive&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libarchive+in%3Apath&type=Code))
 - [x] [CVE-2017-14166](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14166)
 - [ ] [CVE-2017-14501](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14501) [no upstream fix]
 - [x] [CVE-2017-14502](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14502)
 - [ ] [CVE-2017-14503](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14503) [no upstream fix]

### libjpeg-turbo-1.5.2 ([search](https://search.nix.gsc.io/?q=libjpeg-turbo&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libjpeg-turbo+in%3Apath&type=Code))
 - [ ] [CVE-2017-15232](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-15232) [[patch](https://github.com/libjpeg-turbo/libjpeg-turbo/commit/5bc43c7821df982f65aa1c738f67fbf7cba8bd69) doesn't apply cleanly to stable version]

### libsndfile-1.0.28 ([search](https://search.nix.gsc.io/?q=libsndfile&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libsndfile+in%3Apath&type=Code))
 - [x] [CVE-2017-12562](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-12562)
 - [ ] [CVE-2017-14245](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14245) [no upstream fix]
 - [ ] [CVE-2017-14246](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14246) [no upstream fix]
 - [x] [CVE-2017-14634](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14634)
 - [x] [CVE-2017-6892](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-6892)
 - [x] [CVE-2017-8361](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-8361)
 - [x] [CVE-2017-8362](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-8362)
 - [x] [CVE-2017-8363](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-8363)
 - [x] [CVE-2017-8365](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-8365)

### libtasn1-4.12 ([search](https://search.nix.gsc.io/?q=libtasn1&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libtasn1+in%3Apath&type=Code))
 - [x] [CVE-2017-10790](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-10790)

### libtiff-4.0.8 ([search](https://search.nix.gsc.io/?q=libtiff&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libtiff+in%3Apath&type=Code))
 - [x] [CVE-2017-10688](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-10688)
 - [x] [CVE-2017-11335](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-11335)
 - [ ] [CVE-2017-11613](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-11613) [no upstream fix, debian and redhat won't fix this]
 - [x] [CVE-2017-12944](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-12944)
 - [x] [CVE-2017-13726](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-13726)
 - [x] [CVE-2017-13727](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-13727)
 - [ ] [CVE-2017-9935](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-9935) [no upstream fix]
 - [x] [CVE-2017-9936](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-9936)
 - [x] [CVE-2017-9937](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-9937)

### libvorbis-1.3.5 ([search](https://search.nix.gsc.io/?q=libvorbis&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libvorbis+in%3Apath&type=Code))
 - [ ] [CVE-2017-11333](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-11333) [no upstream fix]
 - [ ] [CVE-2017-14160](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14160) [no upstream fix]
 - [ ] [CVE-2017-14632](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14632) [no upstream fix]
 - [ ] [CVE-2017-14633](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14633) [no upstream fix]

### libxslt-1.1.29 ([search](https://search.nix.gsc.io/?q=libxslt&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libxslt+in%3Apath&type=Code))
 - [x] [CVE-2017-5029](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-5029)

### openjpeg-2.1.2 ([search](https://search.nix.gsc.io/?q=openjpeg&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=openjpeg+in%3Apath&type=Code))
 - [x] [CVE-2016-10504](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-10504)
 - [x] [CVE-2016-10505](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-10505)
 - [x] [CVE-2016-10506](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-10506)
 - [x] [CVE-2016-10507](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-10507)
 - [x] [CVE-2016-9112](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-9112)
 - [x] [CVE-2016-9113](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-9113)
 - [x] [CVE-2016-9114](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-9114)
 - [x] [CVE-2016-9115](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-9115)
 - [x] [CVE-2016-9116](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-9116)
 - [x] [CVE-2016-9117](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-9117)
 - [x] [CVE-2016-9118](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-9118)

### openldap-2.4.45 ([search](https://search.nix.gsc.io/?q=openldap&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=openldap+in%3Apath&type=Code))
 - [x] [CVE-2017-14159](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-14159)

### pcre-8.40 ([search](https://search.nix.gsc.io/?q=pcre&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=pcre+in%3Apath&type=Code))
 - [x] [CVE-2017-7186](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-7186)
 - [x] [CVE-2017-7244](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-7244)
 - [x] [CVE-2017-7245](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-7245)
 - [x] [CVE-2017-7246](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-7246)

### perl-5.24.2 ([search](https://search.nix.gsc.io/?q=perl&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=perl+in%3Apath&type=Code))
 - [x] [CVE-2017-12814](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-12814)
 - [x] [CVE-2017-12837](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-12837)
 - [x] [CVE-2017-12883](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-12883)

### sqlite-3.20.0 ([search](https://search.nix.gsc.io/?q=sqlite&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=sqlite+in%3Apath&type=Code))
 - [x] [CVE-2017-13685](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2017-13685)

### owncloud-7.0.5 ([search](https://search.nix.gsc.io/?q=owncloud&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=owncloud+in%3Apath&type=Code))
 - [x] [CVE-2015-4717](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2015-4717)
 - [x] [CVE-2015-4718](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2015-4718)
 - [x] [CVE-2015-5954](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2015-5954)
 - [x] [CVE-2015-6500](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2015-6500)
 - [x] [CVE-2015-6670](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2015-6670)
 - [x] [CVE-2015-7699](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2015-7699)

### jython-2.7.0 ([search](https://search.nix.gsc.io/?q=jython&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=jython+in%3Apath&type=Code))
 - [x] [CVE-2016-4000](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-4000)

### zabbix-2.0.11 ([search](https://search.nix.gsc.io/?q=zabbix&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=zabbix+in%3Apath&type=Code))
 - [x] [CVE-2014-9450](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2014-9450)
 - [x] [CVE-2016-4338](https://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2016-4338)

**Comment by grahamc** [comment]
🚨 ⚡️ 👍 🔥 🚒 🎆 🙌 📯 😄 😂 😹

**Comment by grahamc** [comment]
cc: @LnL7 @7c6f434c @fpletz @globin @NeQuissimus @wizeman @FRidh @vcunat @peterhoeg @ndowens @dezgeg @nh2.

Note: The list of people CC'd on this issue participated in the last
roundup. If you participate on this roundup, I'll cc you on the next
one. If you don't participate in the next one, you won't be CC'd on
the one after that. If you would like to be CC'd on the next roundup,
add a comment to the most recent vulnerability roundup.

Permanent CC's: @joepie91, @phanimahesh, @the-kenny, @7c6f434c, @k0001, @peterhoeg
@NixOS/security-notifications

If you would like to be CC'd on all roundups, PM me.

**Comment by ckauhaus** [comment]
This list needs manual cleanup since vulnix does not honour CVE patches already present. Working on this.

**Comment by disassembler** [comment]
I'm looking at freetype.

**Comment by disassembler** [comment]
freetype-2.7.1 (search, files) (CVE-2017-8105) is already patched.

**Comment by dezgeg** [comment]
Sqlite needs merging https://github.com/NixOS/nixpkgs/pull/30927 AFAICT.

**Comment by disassembler** [comment]
https://github.com/NixOS/nixpkgs/pull/21161 Should resolved jquery-ui CVE.

**Comment by disassembler** [comment]
libxslt is already patched in master and 17.09.

**Comment by disassembler** [comment]
I'll take libjpeg-turbo

**Comment by fpletz** [comment]
### libvorbis

 * CVE-2017-11333
    * No upstream fix yet https://gitlab.xiph.org/xiph/vorbis/issues/2332
    * Only a DoS, Debian considers this a minor issue https://security-tracker.debian.org/tracker/CVE-2017-11333
 * CVE-2017-14160
    * No upstream fix yet https://gitlab.xiph.org/xiph/vorbis/issues/2330
 * CVE-2017-14632
    * No upstream fix yet https://github.com/xiph/vorbis/issues/29
 * CVE-2017-14633
    * No upstream fix yet https://gitlab.xiph.org/xiph/vorbis/issues/2329

**Comment by NeQuissimus** [comment]
All the audiofile stuff seems to be mpruett/audiofile#42

**Comment by pbogdan** [comment]
sqlite - staging is already on 3.20.1 but that suffers from CVE-2017-15286, a PR is opened to update to 3.21.0 https://github.com/NixOS/nixpkgs/pull/30927

**Comment by pbogdan** [comment]
libtiff - we have patches for:

- CVE-2017-9936
- CVE-2017-10688
- CVE-2017-11335
- CVE-2017-12944
- CVE-2017-13726
- CVE-2017-13727

CVE-2017-9937 - appears to be a incorrectly assigned, the bug is in jbigkit package - http://bugzilla.maptools.org/show_bug.cgi?id=2707

**Comment by 7c6f434c** [comment]
`libtasn1`: source reports fixing `CVE-2017-9310` with an upstream patch; in reality it seems to be a patch for `CVE-2017-10790`. It seems that https://nvd.nist.gov/vuln/detail/CVE-2017-9310 is about Qemu anyway… It seems that Debian imported the same patch http://git.savannah.nongnu.org/cgit/libtasn1.git/patch/?id=d8d805e1f2e6799bb2dff4871a8598dc83088a39 as a fix for `CVE-2017-10790` in 
https://anonscm.debian.org/cgit/pkg-gnutls/libtasn1.git/commit/?id=75d2ac4bcc8c92515d1bbda33103c5f78d353a6e

Could anyone recheck that I am not missing something?

**Comment by adisbladis** [comment]
`CVE-2017-14166` and `CVE-2017-14502` (`libarchive`) fixed in https://github.com/NixOS/nixpkgs/pull/30990

**Comment by peterhoeg** [comment]
The following ```openjpeg``` CVEs are already fixed in master and 17.09 ref 428927ffa6e5c255ef97f62435b0777f8f9481df :
- CVE-2016-9112
- CVE-2016-9114
- CVE-2016-9116
- CVE-2016-9118

So I'm marking those.

**Comment by flokli** [comment]
Actually, CVE-2017-10790 got fixed in 1fb803c3677ad4423a68f1520b266c440d06bb23 (although pretending to have fixed CVE-2017-10790), so this can be checked too.

I added a commit in #31082 fixing the patch number.

**Comment by flokli** [comment]
jython fix comes with #31090.

**Comment by vcunat** [comment]
I've got bad news for `nixos-unstable*` channels.

`-small` jobset [has succeeded](https://hydra.nixos.org/eval/1406520) hours ago, but apparently the channel script is stuck.  The big jobset is blocked by [one test](https://hydra.nixos.org/build/63190977) that seems impossible to start, even if I cancel and restart it.  /cc @edolstra

**Comment by vcunat** [comment]
I picked this round of mass rebuilds to 17.09.  In case of e.g. sqlite it may have been more suitable to find and apply only specific patches instead of the upgrades; certainly feel free to do so (I don't have that much time to spare).

**Comment by vcunat** [comment]
And speaking of time, I ignored **17.03**, at least for now.  IIRC we used to leave only a month's overlay, so we *might* just drop picking to 17.03 except for really critical problems.  I really have no idea how fast NixOS users migrate, but one month probably isn't too much.  I believe we should at least explicitly **announce** such plans in advance, saying e.g. that 17.09 shall be maintained (at least) until the end of May 2018.

**Comment by fpletz** [comment]
I was also just about to also push those cherry-picks after having tested a few builds. Thanks! I'm more liberal concerning minor package bumps if they don't break any of our packages.

Regarding the security support: I agree but let's wait until we have the security team and 'on call' figured out. I'll also try to fix the most important ones for 17.03.

**Comment by adisbladis** [comment]
All of the`owcloud` vulnerabilities are fixed after version `8.1.2`.

@matejc Is there any reason to keep so many old versions around?

**Comment by vcunat** [comment]
We have a libc problem: https://nvd.nist.gov/vuln/detail/CVE-2017-15670

Plan:
- [ ] pick just the patch for 17.09, and verify there aren't more missed security patches
- [ ] on staging update to 2.26+stable directly, and get to nixos-unstable* soon

**Comment by grahamc** [comment]
Sounds good, vcunat. Other distro bug trackers and source repos can prove
helpful to enumerate security patches we may need.

**Comment by vcunat** [comment]
glibc: on 17.09 it's in channels (probably), and we have it on master as well, with some caveats: 6ffafc78fbc.

**Comment by fpletz** [comment]
Patched the audiofile issues.

**Comment by fpletz** [comment]
Fixed all remaining fixable CVEs and added comments to the remaining issues. Most can't be fixed because there is no upstream fix yet. I think we can close this issue now and do another roundup soon.
